### PR TITLE
App Logo/Home Page Design

### LINKED ISSUE #48: Design: Homescreen
Choose or design logo/image for our homescreen to brand our app.

### PR DESCRIPTION
**Pull request by djtaylor8**
## Description

-Added new logo, background gradient, and updated buttons on home screen
-Removed Header component
-Created Learn More modal with bullet point walkthrough of the app
-Created a simple form to join an existing list that will render conditionally based on clicking `Join List`

## Related Issue

closes #48 and #37

## Acceptance Criteria

Make it look good!

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### After
Landing page:

Screen Shot 2022-03-02 at 8 35 12 AM

After a user clicks Join List:
Screen Shot 2022-03-03 at 10 22 36 PM

Learn More modal:
Screen Shot 2022-03-03 at 10 23 48 PM

## Testing Steps / QA Criteria
-Pull down `dj-app-logo` to your local environment and `npm start`
-UPDATED: I went ahead and combined the other issue into this PR to keep things simple. Everything is wired up now and should be ready to go. I'm sure there will be adjustments to make for responsiveness - let me know what you find!

### CODE REVIEW

**Review by shelleymcq — APPROVED** [review]
LOVE the way it looks! Much more interesting and landing-pagey. I like the gradient, too, but we should use the same colors throughout. This is our blue [sky-700 (#0369a1)] and green [teal-500 (#14b8a6)].

**Comment by djtaylor8** [comment]
> LOVE the way it looks! Much more interesting and landing-pagey. I like the gradient, too, but we should use the same colors throughout. This is our blue [sky-700 (#0369a1)] and green [teal-500 (#14b8a6)].

Sweet! I'll update the colors.

**Comment by shelleymcq** [comment]
The home page looks awesome! And the way you handled the about and join features is perfect. "automagically" :D

I noticed the header is gone from all pages, not just home. Was that intentional, and we'll put it back when Lindsey is done adding the button features?

**Comment by djtaylor8** [comment]
> The home page looks awesome! And the way you handled the about and join features is perfect. "automagically" :D
> 
> I noticed the header is gone from all pages, not just home. Was that intentional, and we'll put it back when Lindsey is done adding the button features?

Ah, that works. I think the header was only being used on the Home page on my branch which is why I removed it. We can add it back for her!

**Comment by alenamedved** [comment]
Great job, DJ! It looks so beautiful and professional, I love it and feel like I want to use this app.
I've noticed a small thing: when 'join list' button clicked it causes the whole paragraph to 'jump' up, I'm not sure that it is a good user experience, can we just make input appear under buttons without changing buttons position?

File: src/components/Home.jsx [hunk 1]
```diff
@@ -4,10 +4,11 @@ import { collection, addDoc, getDocs, query, where } from 'firebase/firestore';
 import { db } from '../lib/firebase';
 import { getToken } from '@the-collab-lab/shopping-list-utils';
 import toast, { Toaster } from 'react-hot-toast';
+import LearnMoreModal from './LearnMoreModal';
 
 function Home() {
   const [userToken, setUserToken] = useState('');
-
+  const [showJoinListForm, setJoinListForm] = useState(false);
```
Comment (alenamedved) [inline src/components/Home.jsx]: love this approach to make join list input visible only when button was clicked

**Comment by djtaylor8** [comment]
> Great job, DJ! It looks so beautiful and professional, I love it and feel like I want to use this app. I've noticed a small thing: when 'join list' button clicked it causes the whole paragraph to 'jump' up, I'm not sure that it is a good user experience, can we just make input appear under buttons without changing buttons position?

Right, I noticed this too. It is in a flex column, so it effects the other divs as well. Does anyone have any suggestions for how to fix this?

File: src/components/Home.jsx [hunk 2]
```diff
@@ -49,43 +50,92 @@ function Home() {
     }
   };
 
+  const joinListForm = () => {
```
Comment (alenamedved) [inline src/components/Home.jsx:53]: wow, I learned something new, actually now I see how obvious and simple this approach, I just have never used a function like this, I'd just create and  return a new component

File: src/components/Home.jsx (hunk 1 above)
Comment (alenamedved) [inline src/components/Home.jsx]: should it be 'setShowJoinListForm'? or you named it 'setJoinListForm' on purpose

File: src/components/LearnMoreModal.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            you'll be a smart shopper!
+          </li>
+        </ul>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <button
```
Comment (alenamedved) [inline src/components/LearnMoreModal.jsx:39]: do we want to add 'aria-label' ? something like 'learn more button'

File: src/components/LearnMoreModal.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          <div className="justify-center items-center flex ml-0 overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none">
+            <div className="relative w-auto my-6 mx-auto max-w-3xl">
+              <div className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
+                <div className="flex items-start justify-between p-5 border-b border-solid border-blueGray-200 rounded-t">
+                  <h3 className="text-3xl font-semibold text-mostlyBlack">
+                    Welcome to Smart Shopper!
+                  </h3>
+                  <button
+                    className="p-1 ml-auto bg-transparent text-mostlyBlack border-0 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
+                    onClick={() => setShowModal(false)}
```
Comment (alenamedved) [inline src/components/LearnMoreModal.jsx:59]: same here 'aria-label' 'close learn more window' or something like that

File: src/components/LearnMoreModal.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    </span>
+                  </button>
+                </div>
+                <div className="relative p-6 flex-auto">
+                  {renderAboutText()}
+                </div>
+                <div className="flex items-center justify-end p-6 border-t border-solid border-blueGray-200 rounded-b">
+                  <button
+                    className="text-rose-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
+                    type="button"
```
Comment (alenamedved) [inline src/components/LearnMoreModal.jsx:73]: and 'aria-label' 'close learn more window' or something like that

**Review by alenamedved — APPROVED** [review]

**Comment by alenamedved** [comment]
> Right, I noticed this too. It is in a flex column, so it effects the other divs as well. Does anyone have any suggestions for how to fix this?
> 
I don't know yet, but will play with it
upd.
So I've tried and moved it outside flex container so it wont effect it positions, but in this case it appears to far from the buttons. For now I think that grid will work, cause in this case we can fix sizes and positions of each container inside this div
buttoons

**Comment by djtaylor8** [comment]
>Yeah, I tried that as well. The newest commit I just pushed up removes the buttons if a user clicks join list, but the large text still moves up to accommodate space for the form. Do we like this better?
>
I like better with buttons. 
I was able to do it using grid, do you want me to copy it here? @djtaylor8

**Comment by djtaylor8** [comment]
> > Yeah, I tried that as well. The newest commit I just pushed up removes the buttons if a user clicks join list, but the large text still moves up to accommodate space for the form. Do we like this better?
> 
> I like better with buttons. I was able to do it using grid, do you want me to copy it here? @djtaylor8

yes please!

**Comment by alenamedved** [comment]
`const joinListForm = () => {
    return (
      <>
        <form className="w-full max-w-sm mt-6  row-start-4 row-span-2">...`

`return (
    <div className="ml-0 leading-normal tracking-normal text-white h-auto">
      <Toaster />
      <div className="mt-12">
        <div className="items-stretch md:items-center container px-3 mx-auto flex flex-wrap flex-col lg:flex-row items-center justify-between">
          <div className="grid  grid-rows-5 flex-col w-full md:w-1/2 justify-center items-start text-center md:text-left">
          <h1 className="self-end my-4 text-6xl font-bold leading-tight font-Staatliches">
              Smart Shopper
            </h1>...`

line 140
`<div className="w-full md:w-1/2 py-6 text-center order-first lg:order-last">`

hope I did not forget anything
Also I had to put img before text on small screen, as I added empty grids, this way it works

**Comment by djtaylor8** [comment]
> self-end my-4 text-6xl font-bold leading-tight font-Staatliches

This looks good, though I kind of prefer the main logo being centered rather than toward the top of the page. Also, there's some funky stuff happening with the button spacing now too.

**Comment by djtaylor8** [comment]
@shelleymcq @lindseyindev what do ya'll think about this?

**Comment by alenamedved** [comment]
> > self-end my-4 text-6xl font-bold leading-tight font-Staatliches
> 
> This looks good, though I kind of prefer the main logo being centered rather than toward the top of the page. Also, there's some funky stuff happening with the button spacing now too.

oh yes, missed this one, cause 'learn more' button has a margin on the small screen

**Comment by djtaylor8** [comment]
Most recent changes are pushed up now. Gah I LOVE playing with button margins on a Friday afternoon 😂

**Comment by djtaylor8** [comment]
Alena, I added your changes here as well. If everyone is good with it, let's merge!

**Comment by alenamedved** [comment]
the last  changes can be applied only on medium  size screen and keep the small screen untouched, maybe it is the best way

**Comment by djtaylor8** [comment]
> the last changes can be applied only on medium size screen and keep the small screen untouched, maybe it is the best way

which changes?

**Comment by alenamedved** [comment]
> > the last changes can be applied only on medium size screen and keep the small screen untouched, maybe it is the best way
> 
> which changes?

I mean grid container, we can have it on medium and large screens,  and leave small screen flex. This way image will be centered on the small screen

**Comment by djtaylor8** [comment]
> > > the last changes can be applied only on medium size screen and keep the small screen untouched, maybe it is the best way
> > 
> > 
> > which changes?
> 
> I mean grid container, we can have it on medium and large screens, and leave small screen flex. This way image will be centered on the small screen

I kinda like it like this - the image looks good on the small screen. It's pretty big, but that's ok.

**Comment by shelleymcq** [comment]
@djtaylor8 It looks great! Can you try one thing? Can you change the background gradient to end at teal-600 (#0d9488)? That would provide a little contrast for the frames around sections on listView and addItem. And then go ahead and merge with whichever one you think looks best.

**Comment by djtaylor8** [comment]
> @djtaylor8 It looks great! Can you try one thing? Can you change the background gradient to end at teal-600 (#0d9488)? That would provide a little contrast for the frames around sections on listView and addItem. And then go ahead and merge with whichever one you think looks best.

Done!

**Comment by djtaylor8** [comment]
I'm going to push this up to main so we can start tweaking things across the whole app if needed. Feel free to make changes if you see something!

**Pull request merged**